### PR TITLE
Start docker wordpress article

### PR DESCRIPTION
**Pull request by gianarb**

### CODE REVIEW

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+img: /img/docker.png
+tags: devops, docker
+summary: "Docker and wordpress to guarantee scalability, flexibilty and isolation.
+A lot of webagencies install all wordpress in the same server but how can they manage a disaster?
+AWS with Elastic Container Service could be a more professional solution."
+priority: 0.6
+changefreq: yearly
+---
+
+I am trying to rapresent a typically wordpress infrastructure
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: I am trying to represent a typical wordpress infrastructure

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+![Wordpress typically infrastructure](/img/posts/2015-12-16/wp-infra.png)
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: Typical Wordpress infrastructure

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+**Isolation**: every single wordpress share all with the others, filesystem, memory, database.
+This lack of isolation is cause of different problems:
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: This lack of isolation causes different problems:

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+* The monitoring of each installation is more difficult.
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: The monitoring of each installation is harder.

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+* We share security problem
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: problems

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+* We are not freedom to work without fear to block 100 customers
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: We don't have the freedom to work without the fear of blocking 100 customers

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+we submerged by problems and we hide a lot of them.
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: We are overwhelmed by the problems ....

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+
+![Problem](/img/posts/2015-12-16/problem.png)
+
+## LXC Container
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: ATM docker sits on top of runc (https://github.com/opencontainers/runc)

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+> isolated Linux systems (containers) on a single Linux control host.
+>
+> by wikipedia
+
+Wikipedia helps me to resolve one problem (theory), container is **isolated
+Linux System**
+
+## Docker
+
+Docker is a console application that wrap LinuX Containers technology and helps yuo to
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: This sentence doesn't make sense to me

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+delivery container ready to go in production  in a complete filesystem that
+contains everything it needs to run: code, runtime, system tools, system
+libraries.
+
+Worpdress in this implemetation has two containers, one to provide apache and php and one for mysql database.
+This is an example of Dockerfile, it describe how a docker container works it is very simple to understand, from this example there are different keywords
+
+* `FROM` describes the image that we use how start point.
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: describes the image that we use **as** start point.

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+* `RUN` execute and action.
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: run a command

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+* `EXPOSE` describes ports to open during a link, in this case MySql runs on the default port 3306.
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: this is not very clear

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+* `CMD` is the default command used during the run console command.
+
+{% highlight dockerfile %}
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: why not giving them the official mysql image here?

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+use the official [mysql](https://hub.docker.com/_/mysql/) and
+[wordpress](https://hub.docker.com/_/wordpress/) images.
+
+Download this images
+{% highlight dockerfile %}
+docker pull wordpress
+docker pull mysql
+{% endhighlight %}
+
+We are ready to run all! At the moment we are only build or download our
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: ???

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+
+docker run -e WORDPRESS_DB_HOST=wp1.database.prod \
+    -e WORDPRESS_DB_USER=root \
+    -e WORDPRESS_DB_PASSWORD=help_me \
+    -p 8080:80 \
+    -d --name wp1 \
+    --link wp.database.prod:mysql wordpress
+{% endhighlight %}
+
+I can try to examplain this commands, it run two container:
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: explain
runs two containers

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+
+* The name of the first container is mysql and it use `mysql` image, we
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: uses the

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+  use -p flag to expose mysql port now you can use phpmyadmin or other client
+  to fetch the data but remember this is not a good practice.
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: remember that

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+* The second container called wp1 uses the image `gianarb/wordpress` forward
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: our should be 'host' people may not understand what 'our' is for

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+  the container port 80 (apache) on our 8080, in this way we can see the site.
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: exposing the port is not needed to "see the site"

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+  --link flag is the correct way to consume mysql outside the main container,
+  in this particular case we could use wp.database.prod how url to connect at
+  mysql from our worpdress container, awesome!
+* An images can support environment variable in order to buil in the best way
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: ??

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+  the content of container, in this case to set root password in mysql and to
+  configure worpdress's dataase connection
+
+We are ready! Now you have a worpdress ready to go on port 8080.
+
+## Docker Compose
+To save time and to increase reusability we can use [](docker-compose) tool
+that helps us to manage multi-container infrastructures, in this case one for
+mysql and one for wordpress.
+In pracice we can describe all work did above in a `docker-compose.yml` file:
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: typo

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+href="http://giphy.com/gifs/foxtv-win-ricky-gervais-emmys-2015-l41lYCDgxP6OFBruE">via
+GIPHY</a></p>
+</p>
+
+## In Cloud with AWS ECS
+We won a battle but the war is too long, we can not use our PC as server.  In
+this article I propose [AWS Elastic Container
+Service](http://docs.aws.amazon.com/AmazonECS/latest/developerguide/Welcome.html)
+a new AWS service that helps us to manage containers, why this service? Because
+it is Docker and Docker Composer like!
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: Not a real reason to choose that platform

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+
+![AWS Elastic Container Service](/img/posts/2015-12-16/ecs.png)
+
+A services of keywords to understand how it works:
+
+* **Container instance**: An Amazon EC2 that is running the Amazon ECS Agent. It has been registered to ECS Cluster
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: into the ECS...

File: _posts/2015-12-16-wordpress-docker.markdown [hunk 24]
```diff
@@ -0,0 +1,230 @@
...
+
+4. Follow the status of your tasks
+{% highlight bash %}
+ecs-cli ps
+{% endhighlight %}
+
+You can use another docker-compose.yml with a different wordpress port to build another task with another worpdress!
+
+## Now is only a problem of URL
+Ok we are different isolated worpdress on cloud, but they are an ip and different ports..
```
Comment (fntlnz) [inline _posts/2015-12-16-wordpress-docker.markdown]: ??

File: _posts/2015-12-16-wordpress-docker.markdown (hunk 22 above)
Comment (gianarb) [inline _posts/2015-12-16-wordpress-docker.markdown]: Maybe it is not the only reason to choose that platform :hushed:

**Pull request merged**